File: go_src/007-reverse-integer/main.go
package main

import (
	"fmt"
	"math"
)

func reverse(x int) int {
	if x > math.MaxInt32 || x < math.MinInt32 {
		return 0
	}

	res := 0

	// 321 -> 123
	if x >= 0 {
		for ; x != 0; x /= 10 {
			ge := x % 10
			res = 10 * res
			res += ge
		}
	} else {
		x = -x
		for ; x != 0; x /= 10 {
			ge := x % 10
			res = 10 * res
			res += ge
		}
		res = -res
	}

	if res > 2147483647 || res < -2147483648 {
		return 0
	}

	return res
}

func main() {
	res := reverse(1534236469)
	fmt.Println(res)
}

File: go_src/007-reverse-integer/main.py


class Solution:
    def reverse(self, x: int) -> int:

        # if x > 2**31-1 or x < -2**31:
        #     return 0

        if x > 2147483647 or x < -2147483648:
            return 0

        st = str(x)
        answer = 0

        if x > 0:
            answer = int(st[::-1])

        elif x < 0:
            answer = 0 - int(st[:0:-1])

        if answer > 2147483647 or answer < -2147483648:
            return 0
        else:
            return answer


s = Solution()
print(s.reverse(4444444))
